drivers/nuopc/mod_nuopc_methods.F90

```diff
@@ -57,60 +62,239 @@ module mod_nuopc_methods
+        index_Foxx_lwup   = - 1, &
+        index_Foxx_evap   = - 1, &
+        index_Foxx_swnet  = - 1, &
+        index_Sw_lamult   = - 1, &
+        index_Sw_ustokes  = - 1, &
+        index_Sw_vstokes  = - 1, &
+        index_Sw_hstokes  = - 1, &
+        index_Faxa_lwdn   = - 1, &
+        index_Faxa_snow   = - 1, &
+        index_Faxa_rain   = - 1, &
+        index_Sa_pslv     = - 1, &
+        index_Sa_co2diag  = - 1, &
+        index_Sa_co2prog  = - 1
+
+   ! Indices for export fields
+   integer  :: &
+        index_So_omask   = - 1, &
+        index_So_u       = - 1, &
+        index_So_v       = - 1, &
+        index_So_dhdx    = - 1, &
+        index_So_dhdy    = - 1, &
+        index_So_t       = - 1, &
+        index_So_s       = - 1, &
+        index_So_bldepth = - 1, &
+        index_So_dms     = - 1, &
+        index_So_brf     = - 1, &
+        index_Fioo_q     = - 1, &
+        index_Faoo_fco2_ocn = - 1
+
+
+   ! Set logicals for CPP variables
+#ifdef BROMO
+   logical :: use_bromo = .true.
+#else
+   logical :: use_bromo = .false.
+#endif
+#ifdef PROGCO2
+   logical :: progco2 = .true.
+#else
+   logical :: progco2 = .false.
+#endif
+#ifdef DIAGCO2
+   logical :: diagco2 = .true.
+#else
+   logical :: diagco2 = .false.
+#endif
+
 contains
 
    ! ---------------------------------------------------------------------------
    ! Private procedures.
    ! ---------------------------------------------------------------------------
 
-   subroutine getfldindex(fldlist_num, fldlist, stdname, fldindex)
+   subroutine fldlist_add(num, fldlist, stdname, index, ungridded_lbound, ungridded_ubound)
    ! ---------------------------------------------------------------------------
-   ! Get index of field with given standard name. If no field has a matching
-   ! name or a field with matching name has an unassociated data pointer, set
-   ! index to zero.
+   ! Add to list of field information.
    ! ---------------------------------------------------------------------------
 
       ! Input/output arguments.
-      integer, intent(in) :: fldlist_num
-      type(fldlist_type), dimension(:), intent(in) :: fldlist
-      character(len=*), intent(in) :: stdname
-      integer, intent(inout) :: fldindex
+      integer           , intent(inout) :: num
+      type(fldlist_type), intent(inout) :: fldlist(:)
+      character(len=*)  , intent(in)    :: stdname
+      integer           , intent(out)   :: index 
+      integer, optional , intent(in)    :: ungridded_lbound, ungridded_ubound
 
-      ! Local variables.
-      integer :: n
+      ! Local parameters.
+      character(len=*), parameter :: &
+         subname = modname//':(fldlist_add)'
 
-      if (fldindex >= 0) return
+      ! Local variables.
+      integer :: rc
+
+#ifdef HAMOCC
+      ! Set the logical flag do_bgc_aofluxes to false in mo_control_bgc since
+      ! for nuopc/cmeps the dms and bromo fluxes will be computed in the mediator 
+      do_bgc_aofluxes = .false.
+#endif
```

Comment:
Is there any possibility to let the user make this choice? For example, if this could be made a namelist variable (with defalt value .false.) it would be easy to activate the old HAMOCC scheme. But I fear this information is needed before the BLOM/HAMOCC namelists are read in?